domain/consensus/utils/utxo/mutable_utxo_diff.go

```diff
@@ -0,0 +1,164 @@
+package utxo
+
+import (
+	"fmt"
+
+	"github.com/kaspanet/kaspad/domain/consensus/model"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/consensushashing"
+	"github.com/kaspanet/kaspad/domain/consensus/utils/transactionhelper"
+	"github.com/pkg/errors"
+)
+
+type mutableUTXODiff struct {
+	toAdd    utxoCollection
+	toRemove utxoCollection
+
+	immutableReferences []*immutableUTXODiff
+}
+
+// NewMutableUTXODiff creates an empty mutable UTXO-Diff
+func NewMutableUTXODiff() model.MutableUTXODiff {
+	return newMutableUTXODiff()
+}
+
+func newMutableUTXODiff() *mutableUTXODiff {
+	return &mutableUTXODiff{
+		toAdd:    utxoCollection{},
+		toRemove: utxoCollection{},
+	}
+}
+
+func (mud *mutableUTXODiff) ToImmutable() model.UTXODiff {
+	immutableReference := &immutableUTXODiff{
+		mutableUTXODiff: mud,
+		isInvalidated:   false,
+	}
+
+	mud.immutableReferences = append(mud.immutableReferences, immutableReference)
+
+	return immutableReference
+}
+
+func (mud *mutableUTXODiff) invalidateImmutableReferences() {
```

Comment:
Why not call invalidateImmutableReferences for each interface method? Than it'll make it harder to forget to call it if you add another method